packages/marketplace/src/sagas/app-detail.ts

```diff
@@ -15,21 +15,35 @@ import { Action } from '@/types/core'
 import { logger } from '@reapit/utils'
 import { fetchAppById, fetchAppSecretById } from '@/services/apps'
 import { fetchApiKeyInstallationById } from '@/services/installations'
+import { history } from '../core/router'
+
+// by remove '/edit' segment in url
+export const getDetailUrlFromEditDetailUrl = (editDetailUrl: string) => {
+  return editDetailUrl.replace(new RegExp('/edit'), '')
+}
 
 export const appDetailDataFetch = function*({ data }: Action<AppDetailParams>) {
   yield put(appDetailLoading(true))
   try {
-    const { id, clientId } = data
+    const { id, clientId, fetchToEdit } = data
     const appDetailResponse = yield call(fetchAppById, {
       clientId,
       id,
     })
-    if (appDetailResponse?.isWebComponent && appDetailResponse?.installationId) {
-      const apiKeyResponse = yield call(fetchApiKeyInstallationById, {
-        installationId: appDetailResponse.installationId,
-      })
-      appDetailResponse.apiKey = apiKeyResponse?.apiKey || ''
+
+    // redirect to detail page if app's pending revision status is true
+    if (fetchToEdit && appDetailResponse?.pendingRevisions) {
+      const detailAppUrl = getDetailUrlFromEditDetailUrl(window.location.pathname)
+      yield call(history.push, detailAppUrl)
```

Comment:
Can we move this logic to `developer-submit-app.tsx`?
The problem is `appDetailDataFetch` saga should handle only 1 thing which is fetching app detail data, the logic that redirects to detail page if app's pending revision status is true should be handled in `developer-submit-app.tsx` itself to avoid complexity in our saga